engine/src/main/scala/cromwell/engine/workflow/workflowstore/AbortRequestScanningActor.scala

```diff
@@ -0,0 +1,68 @@
+package cromwell.engine.workflow.workflowstore
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{Actor, ActorLogging, ActorRef, Props, Timers}
+import com.google.common.cache.CacheBuilder
+import cromwell.core.{CacheConfig, WorkflowId}
+import cromwell.engine.workflow.workflowstore.AbortRequestScanningActor.{AbortConfig, RunScan}
+import cromwell.engine.workflow.workflowstore.WorkflowStoreActor.{FindWorkflowsWithAbortRequested, FindWorkflowsWithAbortRequestedFailure, FindWorkflowsWithAbortRequestedSuccess}
+import cromwell.engine.workflow.{WorkflowActor, WorkflowManagerActor}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+class AbortRequestScanningActor(workflowStoreActor: ActorRef,
+                                workflowManagerActor: ActorRef,
+                                abortConfig: AbortConfig,
+                                workflowHeartbeatConfig: WorkflowHeartbeatConfig) extends Actor with Timers with ActorLogging {
+  private val cache = CacheBuilder.newBuilder()
+    .concurrencyLevel(1)
+    .expireAfterWrite(1L, TimeUnit.HOURS)
+    .build[WorkflowId, java.lang.Boolean]()
+
+  self ! RunScan
+
+  override def receive: Receive = {
+    case RunScan =>
+      workflowStoreActor ! FindWorkflowsWithAbortRequested(cromwellId = workflowHeartbeatConfig.cromwellId)
+    case FindWorkflowsWithAbortRequestedSuccess(ids) =>
+      val oldIds = cache.getAllPresent(ids.asJava).keySet().asScala
+      // Diff this list to the cache.
+      val newIds = ids.toSet -- oldIds
+      if (newIds.nonEmpty) {
+        // If there are new IDs then query the WorkflowManagerActor for the corresponding ActorRefs.
+        workflowManagerActor ! WorkflowManagerActor.QueryRootWorkflowActorsRequest(newIds)
```

Comment:
ToL: Just curious about the "ask WMA for ActorRef corresponding to WF Id and then send Abort directly to Workflow Actor" compared to "send an `AbortCommand` with this WF Id to the WMA and let it do what it was doing before"

Reply:
hmm let me try that.